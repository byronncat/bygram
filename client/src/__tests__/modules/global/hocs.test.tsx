import { render } from '@testing-library/react'
import rerender from 'react-test-renderer'

import { useGlobalContext, Global, ErrorPage } from '@global'
import { useEffect } from 'react'

describe('hocs', () => {
  describe('Global', () => {
    it('should render the global hoc correctly', async () => {
      const wrapper = rerender.create(<Global>Test</Global>)
      expect(wrapper).toMatchSnapshot()
    })

    it('should refresh the page', async () => {
      const INIT_RENDER_COUNT = 0
      let renderCount = INIT_RENDER_COUNT
      const Test = () => {
        renderCount++ // Increment render count
        const { refreshPage } = useGlobalContext()

        useEffect(() => {
          refreshPage() // Refresh page - Increment render count
        }, [])
        return <div>Test</div>
      }

      await render(
        <Global>
          <Test />
        </Global>
      )
      expect(renderCount).toBe(2)
    })
  })
})
