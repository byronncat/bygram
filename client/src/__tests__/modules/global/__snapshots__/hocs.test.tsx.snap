// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hocs Global should render the global hoc correctly 1`] = `
Array [
  "Test",
  <div
    className="Toastify"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />,
]
`;
